ci/clean-up.yml

```diff
@@ -8,6 +8,12 @@ steps:
     # infra/macos/2-common-box/init.sh:echo "build:darwin --disk_cache=~/.bazel-cache" > ~/.bazelrc
     # infra/vsts_agent_linux_startup.sh:echo "build:linux --disk_cache=~/.bazel-cache" > ~/.bazelrc
 
+    # Linux machines don't seem to recover when this script fails, and they get
+    # renewed by the instance_group
+    if [ "$(uname -s)" == "Linux" ]; then
+        trap "shutdown -h now" EXIT
```

Comment:
I assume shutdown really kills them and doesn’t just terminate them?

Reply:
The `shutdown -h now` command itself just shuts down the machine; the GCP "instance group manager" ensures we always have 10 machines up and running, so if a machine shuts down it will terminate (i.e. delete) it and create a new one (with the same name, but starting with a blank hd).

Reply:
I think there's a delay of a minute or so while the group manager waits to see if the instance comes back on its own, but overall this approach has been successfully working as expected in the startup script.